src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java

```diff
@@ -5169,12 +5170,13 @@ public static Map groupBy(Object[] self, List<Closure> closures) {
      * @return a new Map grouped by keys
      * @since 1.0
      */
-    public static <G, K, V> Map<G, Map<K, V>> groupBy(Map<K, V> self, @ClosureParams(MapEntryOrKeyValue.class) Closure<G> closure) {
-        final Map<G, List<Map.Entry<K, V>>> initial = groupEntriesBy(self, closure);
-        final Map<G, Map<K, V>> answer = new LinkedHashMap<G, Map<K, V>>();
-        for (Map.Entry<G, List<Map.Entry<K, V>>> outer : initial.entrySet()) {
+    public static <G, K, V> Map<G, Map<? extends  K, ? extends V>> groupBy(
+            Map<K, V> self, @ClosureParams(MapEntryOrKeyValue.class) Closure<G> closure) {
```

Comment:
Nevermind my comments on wildcards on self param, the more I think about it that doesn't make sense.

However, wont changing the return type to have wildcards extends mean that nothing can be put in the map that is mapped to the groupBy key which some may rely on being able to do?